sympy/integrals/tests/test_deltafunctions.py

```diff
@@ -35,8 +35,9 @@ def test_deltaintegrate():
     assert deltaintegrate(DiracDelta(x)**2, x) == DiracDelta(0)*Heaviside(x)
     assert deltaintegrate(y*DiracDelta(x)**2, x) == \
         y*DiracDelta(0)*Heaviside(x)
-    assert deltaintegrate(DiracDelta(x, 1)**2, x) is None
-    assert deltaintegrate(y*DiracDelta(x, 1)**2, x) is None
+    assert deltaintegrate(DiracDelta(x, 1)**2, x) == -DiracDelta(0, 2)*Heaviside(x)
```

Comment:
Agree. `deltaintegrate` should be re-factored into manualintegrate now.

I've done this before, and generally find the manualintegrate opaque in its 
current form.  It seems like there should be a class-structure wrapping the 
related functions together.



On Thu, 5 May 2016, Jason K. Moore wrote:

>> @@ -35,8 +35,9 @@ def test_deltaintegrate():
>>      assert deltaintegrate(DiracDelta(x)**2, x) == DiracDelta(0)*Heaviside(x)
>>      assert deltaintegrate(y*DiracDelta(x)**2, x) == \
>>          y*DiracDelta(0)*Heaviside(x)
>> -    assert deltaintegrate(DiracDelta(x, 1)**2, x) is None
>> -    assert deltaintegrate(y*DiracDelta(x, 1)**2, x) is None
>> +    assert deltaintegrate(DiracDelta(x, 1)**2, x) == -DiracDelta(0, 2)*Heaviside(x)
>
> Ok, thanks.
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/sympy/sympy/pull/10923/files/87e80fa7da01805214c52865ea21966dca6bbe6c#r62216685